### ISSUE TITLE
Documentation requests

### ISSUE THREAD

**Issue by cfr42** [sasozivanovic/memoize#10]
Thanks for all the updates, fixes and workarounds! I have a few documentation suggestions

### xparse

The documentation of argument specifiers etc. currently confuses me and I kind-of know vaguely what's going on, which others likely don't. Moreover, there are various degrees of vagueness, but I don't think it is possible to not be confused without consulting `memoize.sty` rather than the documentation.

Pages 110 and 103. 

1. Recategorise `g` and `l` (103) to clarify they're now extensions or say something complicated about the history of `xparse`. Basically, it's confusing to be told these are standard when they no longer are (and when they are no longer documented with `\NewDocumentCommand` and friends). It's also confusing to find `memoize.sty` defining them if they come along with `m`, `o` etc.
3. Remove mention of `u` (103). Alternatively, define it as an extension.
4. Update references to `xparse` documentation e.g. on 102 `xparse` -> `usrguide`. Or at least mention `usrguide`.
5. On 103, does 'set the environment name for this environment' mean 'set the name of the environment to be collected'? I'm getting this from elsewhere in the documentation as I don't really understand the explanation on 103.

### Other

1. Add the index to the table of contents available in the PDF. (My documentation doesn't do this as I couldn't figure out how, but I guess you know as Forest has it.)
2. Include changes in the PDF. (I know where these are, but it's nice if they're in the main documentation.)
3. Give an example for Biblatex e.g. how to extend to `\autocite`, `\autocite*`, `\Autocite`. `\Autocite*`, `\autocites` and `\Autocites`. I currently have

``` latex
\mmzset{%
  begindocument/before/.append style={% biblatex
    auto=\autocite{cite},
    auto=\autocite*{cite},
    auto=\Autocite{cite},
    auto=\Autocite*{cite},
    auto=\autocites{cites},
    auto=\Autocites{cites},
  },
}
```
but I suspect
``` latex
\mmzset{%
  begindocument/before/.append style={% biblatex
    auto=\autocite{cite},
    auto=\Autocite{cite},
    auto=\autocites{cites},
    auto=\Autocites{cites},
  },
}
```
may suffice.

[Why are some of my macros red and others blue?]

**Comment by sasozivanovic** [comment]
Addressed in the pre-release 2c48b1e.

* LaTeX3/`xparse` confusion is hopefully cleared up in the docs. 
* Index added to TOC.
* I added a changelog section to the docs, but only containing the links to the MarkDown changelogs. Sorry, but it's too much work doing this twice, and recompiling the docs for the changelogs.
* All `biblatex` citation commands are automatically submitted to `auto`. Nevertheless, the docs show how to submit new commands, as suggested.

**Comment by cfr42** [comment]
Thank-you! Pointer to changes is fine. I just didn't know where to look (except by looking at git history/diffs).
